pkg/discovery/network/generic_test.go

```diff
@@ -165,3 +284,23 @@ func testDiscoverGenericWith(objects ...runtime.Object) *ClusterNetwork {
 	Expect(err).NotTo(HaveOccurred())
 	return clusterNet
 }
+
+func testDiscoverGenericEmulatedErrWith(expectedErr error, objects ...runtime.Object) *ClusterNetwork {
+	clientSet := fake.NewSimpleClientset(objects...)
+	// Inject error for create services to return expectedErr
+	if expectedErr != nil {
+		clientSet.CoreV1().(*fakecorev1.FakeCoreV1).PrependReactor("create", "services",
+			func(action testing.Action) (handled bool, ret runtime.Object, err error) {
+				return true, nil, expectedErr
+			})
+	} else {
+		clientSet.CoreV1().(*fakecorev1.FakeCoreV1).PrependReactor("create", "services",
+			func(action testing.Action) (handled bool, ret runtime.Object, err error) {
+				return true, &v1.Service{}, nil
+			})
```

Comment:
Since this function is intended to simulate an error, it seems we should expect an error passed in. The `else` here doesn't see to make sense.

Reply:
Deleted nil error case.